codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -546,6 +581,7 @@ void incrementStreamableBytesForTree(int numBytes) {
          */
         private final class Frame {
             final ByteBuf data;
+            final Http2Headers headers;
```

Comment:
I think the `Frame` will either be for `Http2Headers` or `ByteBuf` but not both.  Can we account for this (choose your poison...base classes, templates, interfaces, factories, etc... :) )?